src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/CSharpVirtualDocumentPublisher.cs

```diff
@@ -52,50 +64,63 @@ internal void DocumentManager_Changed(object sender, LSPDocumentChangeEventArgs
 
             if (args.VirtualNew is CSharpVirtualDocumentSnapshot)
             {
-                var csharpContainer = new CSharpVirtualDocumentContainer(args.VirtualNew.Snapshot);
+                var csharpContainer = new CSharpVirtualDocumentContainer(_lazyLspDocumentMappingProvider.Value, args.New, args.VirtualNew.Snapshot);
                 _dynamicFileInfoProvider.UpdateLSPFileInfo(args.New.Uri, csharpContainer);
             }
         }
 
-        // Our IRazorDocumentPropertiesService services as our way to tell Roslyn to show C# diagnostics for files that are associated with the `DiagnosticsLspClientName`.
-        // Otherwise Roslyn would treat these documents as closed and would not provide any of their diagnostics.
-        private sealed class CSharpDocumentPropertiesService : IRazorDocumentPropertiesService
-        {
-            public static readonly CSharpDocumentPropertiesService Instance = new CSharpDocumentPropertiesService();
-
-            private CSharpDocumentPropertiesService()
-            {
-            }
-
-            public bool DesignTimeOnly => false;
-
-            public string DiagnosticsLspClientName => RoslynRazorLanguageServerClientName;
-        }
-
         private class CSharpVirtualDocumentContainer : DynamicDocumentContainer
         {
             private readonly ITextSnapshot _textSnapshot;
+            private readonly LSPDocumentMappingProvider _lspDocumentMappingProvider;
+            private readonly LSPDocumentSnapshot _documentSnapshot;
+            private IRazorSpanMappingService _mappingService;
+            private IRazorDocumentExcerptService _excerptService;
+
+            public override string FilePath => throw new NotImplementedException();
```

Comment:
You sure this isn't called now that we have excerpt & mapping services?

Reply:
I wasn't hitting it, but it can't hurt to implement either. Updated!